shopAPI > product > Display,Product API  추가

LeeYhunHwan opened the pull request:
## display
- 브랜드 목록 조회하기 (getDisplayBrands)
- 브랜드 상세 조회하기 (getDisplayBrandDetail)

## product
- 묶음 배송 상품 목록 조회하기 (getProductsBundle)
- 인기 검색어 조회하기 (getFavoriteKeywords)
- [샵바이 프리미엄 전용] 그룹관리코드 조회하기 (groupManagementCodeInquiry)
- 옵션 목록 조회하기 (getProductOptions)
- [샵바이 프리미엄 전용] 재입고 알림 신청 (requestRestockNotification)
- 상품 검색하기 (getProductsSearch)
- 상품 상세 조회하기 (getProductDetail)
- 베스트 리뷰 상품 검색하기 (getProductBestReview)
- 베스트 셀러 상품 검색하기 (getProductBestSeller)
- 상품번호에 해당하는 모든 전시카테고리 조회하기 (getProductDisplayCategories)
- 옵션 조회하기 (getProductOption)
- 관련 상품 정보 조회하기 (getProductsRelated)
- 상품 번호와 쇼핑몰 번호에 해당하는 단축URL 조회하기 (getProductsUrl)
- 상품에 해당하는 옵션 이미지 목록 조회하기 (getProductOptionsImages)
- 옵션의 이미지 정보 조회하기(getOptionSimages)

LeeYhunHwan committed:
:sparkles: feature: shopAPI > product > Display API  추가

get display brands 브랜드 목록 조회하기
get display brands {brandNo} 브랜드 상세 조회하기

LeeYhunHwan committed:
:hammer: fix: display brands api data 추가

LeeYhunHwan committed:
:sparkles: feat: shopAPI > product > Product API 추가

jerome-geek commented on src/api/product/display.ts:
```diff
@@ -0,0 +1,36 @@
+        filter,
+        pageNumber,
+        pageSize,
+        hasTotalCount,
+        fromDB,
+        sort,
```
default parameter가 빠져있네요

jerome-geek commented on src/api/product/display.ts:
```diff
@@ -0,0 +1,36 @@
+    pageSize?: Number;
+    hasTotalCount?: Boolean;
+    fromDB?: Boolean;
+    sort?: { criterion: String; direction: String };
```
sort안에 criterion, direction은 String으로 선언하는 것보다 enum 타입으로 타입을 강제해주는 게 좋을 것 같아요

jerome-geek commented on src/api/product/display.ts:
```diff
@@ -0,0 +1,36 @@
+            data: { filter, pageNumber, pageSize, hasTotalCount, fromDB, sort },
+        }),
+
+    getDisplayBrandsDetail: (brandNo: String): Promise<AxiosResponse> =>
```
여기도 함수명이 적정하지 않은 것 같네요, brand가 복수형이라면 여러 브랜드의 정보를 조회해야되는데 API와 맞지 않는 것 같네요

jerome-geek commented on src/api/product/index.ts:
```diff
@@ -0,0 +1,4 @@
+import display from './display';
+import product from './product';
```
여기도 상대경로로 쓰는 것보다 baseUrl을 사용해서 작성하는게 일관성있는 코드가 될 것 같은데 어떻게 생각하시나요?

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,251 @@
+const product = {
+    // TODO deliveryTemplateNo을 모름 500 error 발생 추후 테스트 필요
+    getProductsBundle: ({
+        deliveryTemplateNo,
+        hasOptionValues,
+        pageSize,
+        pageNumbewr,
+        productSort,
```
default parameter가 빠졌네요

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,251 @@
+
+import request from 'api/core';
+
+interface ProductsParams {
+    deliveryTemplateNo: Number;
+    hasOptionValues?: Boolean;
+    pageSize?: Number;
+    pageNumbewr?: Number;
+    productSort?: { criterion?: String; direction?: String };
+}
```
API 문서를 참고해보면 nullable이라고 써져있는데, 이러면 type에 null도 추가해주는 것이 좋겠죠?

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,251 @@
+        deliveryTemplateNo,
+        hasOptionValues,
+        pageSize,
+        pageNumbewr,
```
오타 수정해주세요, vscode extension 중에 Code Spell Checker가 있는데 이거 설치해서 spelling 체크를 하시면 더 좋겠네요

LeeYhunHwan replied:
정말 좋은 툴이네요

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,251 @@
+            data: { size },
+        }),
+
+    groupManagementCodeInquiry: (
+        groupManagementCodes: String[],
+        saleStatus: String,
+        isSoldOut: Boolean,
+    ): Promise<AxiosResponse> =>
```
Parameter는 객체 형식으로 묶어주시고, default parameter가 빠졌고,nullable이니 type에 null도 추가해주시고, salesStatus는 enum형식으로 관리하면 좋겠네요

jerome-geek reviewed (changes requested):
전체적으로 중간중간 빠진 부분이 조금 있어서 중간까지만 확인했는데, API 문서 천천히 읽으시면서 처음부터 다시 확인해보시는 게 좋겠네요

LeeYhunHwan committed:
:ok_hand: shopAPI > product > Display API 수정

enum으로 sort 타입 제한, default parameter 추가 및 api 이름 수정

LeeYhunHwan committed:
:ok_hand: feat: shopAPI > product > Product APi 수정

LeeYhunHwan committed:
:hammer: fix: shopAPI > product > product api 수정

LeeYhunHwan committed:
:hammer: fix: shopAPI > product > display API 추가

get 요청 data -> params 로 변경,
상수명 대문자로 변경

LeeYhunHwan committed:
:hammer: fix: shopAPI > product > product API data -> params 로 메서드 수정

jerome-geek commented:
Description 다시 작성해주세요

LeeYhunHwan committed:
:hammer: fix: product API 수정

LeeYhunHwan committed:
:hammer: fix: product api null type -> optional 로 변경

LeeYhunHwan committed:
:hammer: fix: display, product API type 값 수정

LeeYhunHwan committed:
:hammer: fix: accessToken 수정

accessToken: localStorage.accessToken -> accessToken: localStorage.getItem('accessToken') || ''

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,388 @@
+            url: `/products/${productNo}/related-products`,
+        }),
+
+    getProductsUrl: (productNo: string): Promise<AxiosResponse> =>
```
여러 상품의 url을 가져오는게 아니니까 products -> product 단수로 변경하는 것이 좋겠죠?
+ 그냥 url이 아니고 단축 url 이라는 것을 명시하면 더 좋을 것 같은데 어떻게 생각하세요?

jerome-geek replied:
@LeeYhunHwan 이건 왜 함수명을 `getProductUrls` 로 바꾸신걸까요?

LeeYhunHwan replied:
이상하게 수정을 했네요 다시 제대로 수정하겠습니다.

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,388 @@
+        }),
+
+    // TODO productNo, optionNo을 모름 404 error 발생 추후 테스트 필요
+    getOptionsImages: (
```
특정 옵션의 이미지를 조회하는 것 같은데 options를 복수로 쓰면 안되지 않을까요?

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,388 @@
+            }),
+        }),
+
+    groupManagementCodeInquiry: ({
```
여기도 동사가 빠졌네요

jerome-geek commented on src/api/product/product.ts:
```diff
@@ -0,0 +1,388 @@
+        }),
+
+    // TODO productNo을 모름 404 error 발생 추후 테스트 필요
+    getProductOption: (productNo: string): Promise<AxiosResponse> =>
```
한 상품에 옵션이 여러개가 있을 수 있으니 options가 낫겠죠? 실제로 response도 배열형태로 오는 것 같네요

LeeYhunHwan committed:
:hammer: fix: product API 함수명 변경

LeeYhunHwan committed:
:hammer: fix: product API 함수명 수정

getProductUrls -> product.getShortUrl()
